libs/pmp-api/api-service/repository/domain/src/lib/read-models/base-repository-statistics.read-model.ts

```diff
@@ -5,13 +5,15 @@ import {
 } from '@pimp-my-pr/pmp-api/api-service/repository/domain';
 
 export abstract class BaseRepositoryStatisticsReadModel {
```

Comment:
Is ReadModel something like DTO or it's something else?

Reply:
Yes, it is.

Reply:
These days, where we try to avoid adding constraints that entangle clients and servers, we're more likely to use a representation of a message (bytes) to cross a process boundary. (What's the difference? we send data, but we don't send the getters and setters over the wire). So the "DTO" becomes the object representation of those bytes in this process.

"Read models" conceptually belong in the core -- they are the portions of our domain model that are written to provide query support. They arise from the logical bifurcation of the domain model into query responsibilities and mutation responsibilities.

A way to recognize the difference -- if your data is in language primitives (bytes, ints, Strings), then you are probably looking at a DTO. If you are looking at domain concepts (account numbers, amounts, SKUs) then you are more likely in the read model.

(That's not perfect, of course, there are a lot of "anemic" and primitive obsessed domain models out there).

Reply:
For future questions, please create a query issue and approve the pull request to not block it if the question is not directly related to the question.

Reply:
@MaciejSikorski Oki, thanks :)